### PR TITLE
Add MariaDB test

### PR DESCRIPTION
**Pull request by justindbaur**
## 🎟️ Tracking

Relates to https://bitwarden.atlassian.net/browse/PM-13473.

## 📔 Objective

Add MariaDB to the test pipeline so we can check that the migrations we write for MySQL also work for MariaDB. This is expected to fail until we merge #4982.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/test-database.yml [hunk 1]
```diff
@@ -70,6 +70,11 @@ jobs:
           docker compose --profile mssql --profile postgres --profile mysql up -d
         shell: pwsh
 
+      - name: Add Maria DB
```
Comment (withinfocus) [inline .github/workflows/test-database.yml:73]:
```suggestion
      - name: Add MariaDB for Unified
```

File: .github/workflows/test-database.yml [hunk 2]
```diff
@@ -70,6 +70,11 @@ jobs:
...
+        # Use a different port than MySQL
+        run: |
+          docker run --detach --name mariadb --env MARIADB_ROOT_PASSWORD=mariadb-password -p 4306:3306 mariadb:latest
```
Comment (withinfocus) [inline .github/workflows/test-database.yml:76]: ❓ Should we use 10 like our docs have?

File: .github/workflows/test-database.yml [hunk 3]
```diff
@@ -130,13 +141,20 @@ jobs:
           # Default Sqlite
           BW_TEST_DATABASES__3__TYPE: "Sqlite"
           BW_TEST_DATABASES__3__CONNECTIONSTRING: "Data Source=${{ runner.temp }}/test.db"
+          # Maria DB
```
Comment (withinfocus) [inline .github/workflows/test-database.yml:144]:
```suggestion
          # Unified MariaDB
```

File: .github/workflows/test-database.yml (hunk 1 above)
Comment (justindbaur) [reply .github/workflows/test-database.yml:73]: I've always been told it's supposed to be lowercase... but will change it.

File: .github/workflows/test-database.yml (hunk 2 above)
Comment (justindbaur) [reply .github/workflows/test-database.yml:76]: Yes, probably, but the architecture of these tests means we could test multiple versions if we wanted to.

**Review by withinfocus — APPROVED** [review]

**Pull request merged**